src/analysis/retail/monk/shared/hero/MasterOfHarmony/talents/AspectOfHarmonyBase.tsx

```diff
@@ -0,0 +1,129 @@
+import SPELLS from 'common/SPELLS';
+import { TALENTS_MONK } from 'common/TALENTS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, {
+  ApplyBuffEvent,
+  ApplyDebuffEvent,
+  DamageEvent,
+  EventType,
+  HealEvent,
+} from 'parser/core/Events';
+import Combatants from 'parser/shared/modules/Combatants';
+
+export interface CastInfo {
+  totalDamage: number;
+  overkill: number;
+  totalHealing: number;
+  overhealing: number;
+  numBuffs: number;
+  numDebuffs: number;
+  startTime: number;
+}
+
+class AspectOfHarmonyBaseAnalyzer extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+  castEntries: CastInfo[] = [];
+  protected combatants!: Combatants;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS_MONK.ASPECT_OF_HARMONY_TALENT);
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.ASPECT_OF_HARMONY_BUFF),
+      this.onBuffApply,
+    );
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.ASPECT_OF_HARMONY_HOT),
+      this.onPeriodicApply,
+    );
+    this.addEventListener(
+      Events.applydebuff.by(SELECTED_PLAYER).spell(SPELLS.ASPECT_OF_HARMONY_DOT),
+      this.onPeriodicApply,
+    );
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER), this.onHeal);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER), this.onDmg);
+  }
+
+  initEntry(
+    event: ApplyBuffEvent | ApplyDebuffEvent | HealEvent | DamageEvent,
+  ): CastInfo | undefined {
+    // normal init case is for aspect buff apply or for catching pre-pull TFT/CB
+    if (event.ability.guid === SPELLS.ASPECT_OF_HARMONY_BUFF.id || !this.castEntries.length) {
+      this.castEntries.push({
+        totalDamage: 0,
+        overkill: 0,
+        totalHealing: 0,
+        overhealing: 0,
+        numBuffs: 0,
+        numDebuffs: 0,
+        startTime: event.timestamp,
+      });
+    }
+    return this.castEntries.at(-1);
+  }
+
+  onBuffApply(event: ApplyBuffEvent) {
+    this.initEntry(event);
+  }
+
+  onPeriodicApply(event: ApplyBuffEvent | ApplyDebuffEvent) {
+    const entry = this.initEntry(event);
+    if (entry) {
+      if (event.type === EventType.ApplyBuff) {
+        // dont care about buffs on pets
+        if (this.combatants.getEntity(event)) {
+          entry.numBuffs += 1;
+        }
+      } else {
+        entry.numDebuffs += 1;
+      }
+    }
+  }
+
+  onHeal(event: HealEvent) {
+    // coalesence doesn't buff Aspect HOT
```

Comment:
I don't see a reason why we would want to ignore the direct healing/damage from the hot/dot when totaling a cast entry. Is this a copy/paste mishap from Coalesce.tsx?

Reply:
It was essentially to measure the overheal of coalesence to make sure that you weren't wasting the buff and the player was forcing overhealing onto a target solely for coalesence, but open to removing it if you think excluding it is unnecessary.

Reply:
I think the direct healing is significant enough that this should either account for both or we should account for both in some way. High overheal on the hot can be an indicator that you're not spreading AoH well